packages/errors/src/error.ts

```diff
@@ -24,6 +24,7 @@ type SolanaErrorCodedContext = Readonly<{
 
 export class SolanaError<TErrorCode extends SolanaErrorCode = SolanaErrorCode> extends Error {
     readonly context: SolanaErrorCodedContext[TErrorCode];
+    readonly cause?: TErrorCode extends SolanaErrorCodeWithCause ? SolanaError : unknown;
```

Comment:
I don't know that this is enough considering the tests are now failing to assert on the `cause` attribute.

I think we need to explicitly set the `cause` property on the constructor instead of expected the parent constructor to set it for us.

Reply:
Oh, oops. Yes; I meant for those to be type annotations only.

Reply:
Mmmm… weird. No, that should have worked.

https://www.typescriptlang.org/play/?#code/MYGwhgzhAEDCD28QFEBOr6ugUwB4BdsA7AExjQywG8Aoae6VbME+IkAT2mDAFcJsAfgBc0IrwC2AI2xYAPtF6lsAMwCWRbCQDcdBsDYR8qXsHyYAFADobYVAHMIASmi0G76BF4AHWddsOznr0AL40YTQGREbQYNAAvGLYAO5wiCjolgDkUixZADSuPPzYogCMAEwAzCFOulEQSNhWIPD2FmBWxQJOQA

Reply:
Uh, the TypeScript compiler literally produces the same JavaScript with or without that line.

Reply:
Oof. https://github.com/egoist/tsup/issues/1104

Reply:
😢 https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier